frostsnapp/native/src/coordinator.rs

```diff
@@ -31,35 +36,41 @@ pub struct FfiCoordinator {
     pending_for_outbox: Arc<Mutex<VecDeque<CoordinatorSend>>>,
     thread_handle: Mutex<Option<JoinHandle<()>>>,
     keygen_stream: Arc<Mutex<Option<StreamSink<CoordinatorToUserKeyGenMessage>>>>,
-    signing_state: Arc<Mutex<Option<SigningState>>>,
-}
-
-struct SigningState {
-    stream: StreamSink<CoordinatorToUserSigningMessage>,
-    session: SigningDispatcher,
-}
-
-impl Default for FfiCoordinator {
-    fn default() -> Self {
-        Self::new()
-    }
+    signing_session: Arc<Mutex<Option<SigningSession>>>,
+    db: Arc<Mutex<LlsDb<File>>>,
+    core_persist: IndexHandle<PersistCore>,
+    device_names: IndexHandle<llsdb::index::BTreeMap<DeviceId, String>>,
 }
 
 impl FfiCoordinator {
-    pub fn new() -> Self {
-        let coordinator = Arc::new(Mutex::new(FrostCoordinator::new()));
+    pub fn new(mut db: LlsDb<File>) -> anyhow::Result<Self> {
         let manager = Arc::new(Mutex::new(UsbSerialManager::new(Box::new(DesktopSerial))));
         let pending_for_outbox = Arc::new(Mutex::new(VecDeque::new()));
 
-        Self {
-            coordinator,
+        let (core_persist, device_names, coordinator) = db
+            .execute(|tx| {
+                let persist = PersistCore::new(tx)?;
+                let (handle, api) = tx.store_and_take_index(persist);
+                let coordinator = api.core_coordinator()?;
+                let device_names_list = tx.take_list("device_names")?;
+                let device_names_btree = llsdb::index::BTreeMap::new(device_names_list, &tx)?;
+                let device_names_btree_handle = tx.store_index(device_names_btree);
```

Comment:
Curious as to why we add the `device_names` entry here and not in `PersistCore`, is the distinction that we're more willing to forget it?

Reply:
Good question. I didn't leave a comment answering it because I felt it's something that will change quickly. But since you asked, there are two reasons:

1. PersistCore stores `frostsnap_core` stuff which I felt should be separate.
2. There is a problem with doing this because rust only includes [`RefMut::map_split`](https://doc.rust-lang.org/std/cell/struct.RefMut.html#method.map_split) which only allows you to split a `RefMut` into 2! But adding a third field I'd need to split it into three to keep it in line with the design of llsdb. I found out about this limitation while trying to do this so I didn't try and fix it just now (it is fixable with `unsafe` code in llsdb I think).

The long term vision of llsdb is to just have a single struct that you put a derive on and everything happens. It's the next thing I'll work on in my off hours.